shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builtin/yaml/config/YamlUserConfigurationConverter.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.metadata.auth.builtin.yaml.config;
+
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.shardingsphere.infra.metadata.auth.builtin.yaml.swapper.UserYamlSwapper;
+import org.apache.shardingsphere.infra.metadata.auth.model.user.ShardingSphereUser;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.stream.Collectors;
+
+/**
+ * Configuration converter for YAML User content.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class YamlUserConfigurationConverter {
+    private static UserYamlSwapper userYamlSwapper = new UserYamlSwapper();
+
+    /**
+     * convert users Yaml content.
+     * @param users collection of ShardingSphereUser
+     * @return collection of formatting users content
+     */
+    public static Collection<String> convertYamlUserConfigurationFormattings(final Collection<ShardingSphereUser> users) {
```

Comment:
It is necessary? BTW, do you think we can combine `YamlUserConfiguration` and `YamlUser` together? I guess it will be a little confused for these two concept.

Reply:
i think it is necessary. because I think register center must be keep pace with `server.yaml` where  `users` configuration is `username@hostname:password`  ,so I format YamlUserConfiguration to `username@hostname:password`

Reply:
So YamlEngine.marshal(YamlUserConfigurationConverter.convertYamlUserConfigurations(users)` can not work well?

Reply:
Yes,`YamlEngine.marshal(YamlUserConfigurationConverter.convertYamlUserConfigurations(users)` will print 
`
!!map
- !!org.apache.shardingsphere.infra.metadata.auth.builtin.yaml.config.YamlUserConfiguration
  hostname: '%'
  password: root
  username: root
- !!org.apache.shardingsphere.infra.metadata.auth.builtin.yaml.config.YamlUserConfiguration
  hostname: ''
  password: sharding
  username: sharding
`

Reply:
and I find the method `org.apache.shardingsphere.infra.yaml.engine.YamlEngine#marshal`  default serialize `Object`  as `map` , 
![image](https://user-images.githubusercontent.com/34728144/113466291-3535da80-946d-11eb-8a5d-c4c180b8554f.png)
when I use this method marshal the users configuration (Collection<String> users), it contains root tag `!!map` like 
![image](https://user-images.githubusercontent.com/34728144/113466365-da50b300-946d-11eb-8ef4-531a399b1d94.png)
,so can you think about it is necessary overload the method `org.apache.shardingsphere.infra.yaml.engine.YamlEngine#marshal`   like 
![image](https://user-images.githubusercontent.com/34728144/113466528-41bb3280-946f-11eb-9d58-1a9c78d6acb5.png)
and input `rootTag` as `null` where call this method

Reply:
Hi, 

Sorry for my late reply.
Is there any reference of `YamlUserConfigurationConverter.convertYamlUserConfigurations(users)` in product coding?
If it is unnecessary to remain `YamlUserConfigurationConverter.convertYamlUserConfigurations(users)`, we can consider renaming `convertYamlUserConfigurationFormattings` to `convertYamlUserConfiguration`.
If it is still needed and just for zookeeper, do you think it is possible to move this function to `registryCenter` to keep  `YamlUserConfigurationConverter.java` has symmetric converter functions? Or the last way for us is to rename it as `convertYamlUserConfiguration1`, like `Collections2`.

Reply:
Thank you very much for your kindly advice!
There are no any reference of `YamlUserConfigurationConverter.convertYamlUserConfigurations(users)`. OK,I will renaming 
 `convertYamlUserConfigurationFormattings ` to `convertYamlUserConfigurations`.